spec/models/gym_membership_spec.rb

```diff
@@ -21,4 +21,24 @@
       end
     end
   end
+
+  describe 'delegations' do
+    describe '#address and #address_details' do
+      it { should delegate_method(:address).to(:find_gym) }
+      it { should delegate_method(:address_details).to(:find_gym) }
+    end
+  end
+
+  describe 'instance methods' do
+    describe '#address and #address_details', :vcr do
+      let!(:yelp_gym) { GymFacade.find_gym('gHmS3WIjRRhSWG4OdCQYLA') }
+      let(:user) { user_with_gym_membership(yelp_gym_id: yelp_gym.yelp_gym_id) }
+      let(:gym_membership) { user.gym_memberships.first }
+
+      it 'can return the address and address details of the gym membership' do
+        expect(gym_membership.address).to eq yelp_gym.address
+        expect(gym_membership.address_details).to eq yelp_gym.address_details
+      end
+    end
+  end
```

Comment:
I think this works for now, but we should probably remove the delegation and instead use a regular method.  

For future development, this would be a fantastic opportunity to incorporate caching (I also think this would be great for the gym search too!).  In particular with this case, we'll need to make at least two Yelp API calls: one call to get the `address` and a second call to get the `address_details`.  

One possible solution to use only one API call (without caching) would be to return a hash with the `address` and `address_details`.

However, using caching would be a great exploration.  Caching would also allow us to remove the `gym_name` column from GymMembership and just cache that along with the other Gym details.

Not a priority now, but it got my gears spinning haha!